platform-api/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/FactoryService.java

```diff
@@ -209,6 +210,39 @@ public Factory saveFactory(Iterator<FileItem> formData,
     }
 
     /**
+     * Save factory to storage.
+     *
+     * @param factory
+     *         instance of factory which would be stored
+     * @return decorated factory instance of which has been stored
+     * @throws org.eclipse.che.api.core.ApiException
+     *         {@link org.eclipse.che.api.core.BadRequestException} when factory is null
+     *         {@link org.eclipse.che.api.core.ConflictException} when problem occurs during factory storing
+     *         {@link org.eclipse.che.api.core.ServerException} when internal server error occurs
+     */
+    @RolesAllowed("user")
+    @POST
+    @Consumes({MediaType.APPLICATION_JSON})
+    @Produces({MediaType.APPLICATION_JSON})
+    public Factory saveFactory(Factory factory) throws ApiException {
+        final User user = EnvironmentContext.getCurrent().getUser();
+        if (user == null || user.getName() == null || user.getId() == null) {
```

Comment:
User can't be `null` as long as `@RollesAllowed("user")` is present on top of this method